File: Rust/bubble-sort.rs
fn bubble_sort(mut vec: Vec<i32>) -> Vec<i32> {
    loop {
        let mut swapped = false;
        let mut i = 0;
        while i < vec.len() {
            if i >= vec.len()-1 { break; }
            if vec[i] > vec[i+1] {
                swapped = true;
                let value = vec[i];
                vec.remove(i);
                vec.insert(i+1, value);
                break;
            }
            i += 1;
        }
        if !swapped { break; }
    }
    vec
}

File: Ruby/BubbleSort.rb
def bubble_sort(array)
    n = array.length
    swapped = true
    while swapped do
      swapped = false
      (n - 1).times do |i|
        if array[i] > array[i + 1]
          array[i], array[i + 1] = array[i + 1], array[i]
      swapped = true
        end
      end
    end
    array 
 end